highfidelity/hifi
# Fade dissolve effect  JUST AVATARS

This PR extends the dissolve fade entity appear events introduced in PR #10964 to voxel, primitive shape and polyline entities.

### Limitations
At this time, models, voxels, primitives and polylines will potentially have a fade effect applied to them when built with the Create menu. The code where the fade event is hooked on doesn't discriminate between entity creation due to domain loading and entity creation due to user interaction with the Create menu, for instance.

# TEST PLAN 01
1. Check that each model based entity that appears in the world has a correct dissolve fade event applied to it, especially with a scale matching the dissolve effect of model entities.
2. When switching back and forth between to worlds, the model based entity should have a fade in event 3. Check that each voxel based entity that appears in the world has a correct dissolve fade event applied to it, especially with a scale matching the dissolve effect of model entities.
4. When switching back and forth between to worlds, the voxel based entity should have a fade in event triggered each time (no fade out is expected).
5. Check that each primitive based entity (box, sphere, etc...) that appears in the world has a correct dissolve fade event applied to it, especially with a scale matching the dissolve effect of model entities.
6. When switching back and forth between to worlds, the primitive based entity should have a fade in event triggered each time (no fade out is expected).
7. Primitives of same color should fade independently (this may be difficult to visualize but can happen based on network load)
8. Check that each polyline based entity that appears in the world has a correct dissolve fade event applied to it, especially with a scale matching the dissolve effect of model entities.
9. When switching back and forth between to worlds, the polyline based entity should have a fade in event triggered each time (no fade out is expected).
10. Check that each avatar that appears in a location has a correct fade event applied to it, different from the model / voxel / shape / polyline based entities.
11. Check that each avatar (except your avatar) that leaves a location has a correct fade event applied to it that is the inverse of the "avatar appear" event.

This PR should be tested on various GPU (Intel, AMD and NVidia) platforms.
